atlp-rwanda/e-comm-team-amigos-bn
# #184581187 Ft View Specific Item

#### What does this PR do?

This PR allow user to get or view a specific item. It also prevents vendors from viewing items which do not belong to them.

#### Description of Task to be completed?

Have the following endpoints working

- `GET /product/<product-id>`

#### How should this be manually tested?

1. In your **e-comm-team-amigos-bn** directory, open terminal and run `git fetch origin pull/32/head:ft-view-specific-item-184581187`
2. Checkout to **ft-view-specific-item-184581187** branch
3. Run `npm install`
4. Run `npm run dev`
5. In your working directory terminal, run migrations and seeders. **NB:** First run user seeder. replace `userId values`  in product seeder with `IDs` newly created users. make sure that the `IDs` you choose are of **vendor** users.
6. In postman or documentation do the following
     - Make a request on `GET /product/getAllItems`
     - Pick one product id from results in above request, and make a request withit on `GET /product/<product-id>` without user logged in, you should receive the item.
     - Again, signing with a vendor user, and try to get item which does not belong to that vendor, you should receive and error message

#### Any background context you want to provide?

- Have Node Js installed in your local machine
- Should have express server running
- Should have your `.env` file with neccessary variables as in `.env.example`
     
#### What are the relevant pivotal tracker/Trello stories?

Story ID: [184581187 ](https://www.pivotaltracker.com/story/show/184581187)

#### Screenshots

`Test Coverage`

![Screenshot (1485)](https://user-images.githubusercontent.com/55640083/226912915-1e160956-7a07-4b5e-a633-a45639557603.png)

`Documentation`

![Screenshot (1483)](https://user-images.githubusercontent.com/55640083/227015306-3b3314a9-7f70-4a45-8bc4-4c73920e742d.png)
